packages/corejs/src/monitoring/index.tsx

```diff
@@ -0,0 +1,130 @@
+*/
+export function MonitoringProvider({ children, config }: MonitoringConfigType) {
+  useEffect(() => {
+    Sentry.init({
+      dsn: (config.dsn || Constants.manifest.extra.SENTRY_DSN) as string,
+      environment: (config.environment || Constants.manifest.extra.ENV) as string,
+      debug: config.debug || Constants.manifest.extra.ENV !== 'production',
+      tracesSampleRate: config.tracesSampleRate || 0.5,
+      integrations: [
+        new BrowserTracing({ tracePropagationTargets: config.tracingOrigins }),
+      ],
+    });
+  }, []);
+
+  /**
+   * Updates user context information for future events.
+   *
+   * @param id {string} set user for in sentry
+  */
+  const setMonitoringUser = useCallback((id: string): void => {
+    Sentry.setUser({ id });
+  }, []);
+
+  /**
+   * Set key:value that will be sent as tags data with the event.
+   *
+   * Can also be used to unset a tag, by passing `undefined`.
+   *
+   * @param key String key of tag
+   * @param value Value of tag
+  */
+  const setMonitoringTag = useCallback((key: string, value: Primitive): void => {
+    Sentry.setTag(key, value);
+  }, []);
+
+  /**
+   * Error handler function which is used to capture errors in sentry.
+   *
+   * @param error {Error | string} - Caught error that to be send to Sentry.io
+   */
+  const errorHandler = useCallback((error: Error | string): string | null => {
+    if (!Sentry) {
+      return null;
+    }
+
+    return Sentry.captureException(error);
+  }, []);
+
+  /**
+   * Measure the performance of application based on functionality and operation based on it.
+   * Return type of the function is the IIFE, which will helps to close the transaction and complete the measurement.
+   *
+   * @param name {string} - Name of transaction
+   * @param operation {string} - Operation of transaction to be performed
+   * @param data {{[key: string]: number | string}} - Data to be added on transaction
+   */
+  // eslint-disable-next-line @typescript-eslint/ban-types
+  const measurePerformance = useCallback((name: string, op: string, data?: { [key: string]: number | string }): Function => {
+    // This will create a new Transaction
+    const transaction = Sentry.startTransaction({ name, data, op });
+
+    // Set transaction on scope to associate with errors and get included span instrumentation
+    // If there's currently an unfinished transaction, it may be dropped
+    Sentry.getCurrentHub().configureScope((scope) => {
+      scope.setSpan(transaction);
+    });
+
+    return () => {
+      transaction.setStatus(SentryTransactionStatus);
+      transaction.finish();
+    };
+  }, []);
+
+  /**
+   * Set the custom measurement on particular transaction
+   *
+   * @param transactionName Name of the transaction
+   * @param name Name of the measurement
+   * @param value Value of the measurement
+   * @param unit Unit of the measurement. (Defaults to an empty string)
```

Comment:
make the unit param optional in the JSDoc

Reply:
Done